src/selectors/entities/posts.js

```diff
@@ -284,19 +292,21 @@ export function makeGetPostsAroundPost() {
 export function makeGetPostsForThread() {
     return createSelector(
         getAllPosts,
-        (state, props) => props,
-        (posts, {rootId}) => {
+        (state, {rootId}) => state.entities.posts.postsInThread[rootId] || [],
+        (state, {rootId}) => state.entities.posts.posts[rootId],
+        (posts, postsInThread, rootPost) => {
             const thread = [];
 
-            for (const id in posts) {
-                if (posts.hasOwnProperty(id)) {
-                    const post = posts[id];
+            if (rootPost) {
+                thread.push(rootPost);
+            }
 
-                    if (id === rootId || post.root_id === rootId) {
-                        thread.push(post);
-                    }
+            postsInThread.forEach((id) => {
```

Comment:
why not for (const id in postsInThread)?

Reply:
Any particular reason for favouring `for in` instead of `forEach`? From my personal experience number of things that might screw up `for in` outweigh any benefits, it provides .https://stackoverflow.com/questions/500504/why-is-using-for-in-with-array-iteration-a-bad-idea?lq=1.

Reply:
I'd be fine with (for let i = 0; i < arr.length; ...) too, but .forEach syntax is really unnecessary, and makes code harder to read

Reply:
We use forEach all over the place, and personally I prefer it here